addons/l10n_sa/models/account_chart_template.py

```diff
@@ -0,0 +1,43 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, api
+
+
+class AccountChartTemplate(models.Model):
+    _inherit = 'account.chart.template'
+
+    def _prepare_all_journals(self, acc_template_ref, company, journals_dict=None):
+        """ If Saudi Arabia chart, we add 3 new journals Tax Adjustments, IFRS 16 and Zakat"""
+        if company.country_id.code == 'SA':
+            if not journals_dict:
+                journals_dict = []
+            journals_dict.extend(
+                [{'name': 'Tax Adjustments', 'company_id': company.id, 'code': 'TA', 'type': 'general',
+                  'favorite': True, 'sequence': 1},
+                 {'name': 'IFRS 16 Right of Use Asset', 'company_id': company.id, 'code': 'IFRS', 'type': 'general',
+                  'favorite': True, 'sequence': 10},
+                 {'name': 'Zakat', 'company_id': company.id, 'code': 'ZAKAT', 'type': 'general', 'favorite': True,
+                  'sequence': 10}])
+        return super()._prepare_all_journals(acc_template_ref, company, journals_dict=journals_dict)
+
+    def _load_template(self, company, code_digits=None, account_ref=None, taxes_ref=None):
+        account_ref, taxes_ref = super(AccountChartTemplate, self)._load_template(company=company,
+                                                                                  code_digits=code_digits,
+                                                                                  account_ref=account_ref,
+                                                                                  taxes_ref=taxes_ref)
```

Comment:
You should put a condition here to ensure we only run that when installing Saudi Arabia's CoA. Again, using the xmlid.

Reply:
done, I realized this mistake is also present in l10n_ae, i'll do a pr for it and tag you